jitbuilder/control/Jit.cpp

```diff
@@ -187,7 +183,23 @@ internal_initializeJit()
 int32_t
 internal_compileMethodBuilder(TR::MethodBuilder *m, void **entry)
    {
-   return m->Compile(entry);
+   auto rc = m->Compile(entry);
+
+#if defined(J9ZOS390)
+   struct FunctionDescriptor
+   {
+      uint64_t environment;
+      void* func;
+   };
```

Comment:
@mstoodle FYI this is the current simplest way I know of solving the function descriptor issue. I suspect AIX has similar problems. There is likely a much better and more robust way of doing this but it will involve a proper design from someone who understands JitBuilder and it's APIs much better than I do. This does work in all cases I've tested however.